internal/internal.go

```diff
@@ -237,6 +240,15 @@ func (p *envoyExtAuthzGrpcServer) listen() {
 		addr = "grpc://" + addr
 	}
 
+	if p.cfg.EnablePrometheusMetrics {
+		summaryAuthzDuration := prometheus.NewSummary(prometheus.SummaryOpts{
+			Name: "grpc_authz_request_duration_seconds",
```

Comment:
Can you please help me understand the reason to use `NewSummary` v/s something like `SummaryVec` or `HistogramVec`? The latter would be useful if we want aggregates afaik. Similar to [this](https://sourcegraph.com/github.com/open-policy-agent/opa@d718975b5de4b05cf2c88d5a52efaa33e71b0f0e/-/blob/internal/prometheus/prometheus.go?L43). 

Nit: Can we change `grpc_authz_request_duration_seconds` to `grpc_request_duration_seconds`

Reply:
NewSummary is more simple. It's a single reason. I will change to SummaryVec if you want.

Reply:
If we have multiple instances of the plugin and we need to aggregate the results, will we be able to do this?

Also is there any impact from the change in default behavior of Summary as indicated in this comment? It would be helpful if we have more tests in any case.
 ```
// A Summary captures individual observations from an event or sample stream and
// summarizes them in a manner similar to traditional summary statistics: 1. sum
// of observations, 2. observation count, 3. rank estimations.
//
// A typical use-case is the observation of request latencies. By default, a
// Summary provides the median, the 90th and the 99th percentile of the latency
// as rank estimations. However, the default behavior will change in the
// upcoming v1.0.0 of the library. There will be no rank estimations at all by
// default. For a sane transition, it is recommended to set the desired rank
// estimations explicitly.
//
// Note that the rank estimations cannot be aggregated in a meaningful way with
// the Prometheus query language (i.e. you cannot average or add them). If you
// need aggregatable quantiles (e.g. you want the 99th percentile latency of all
// queries served across all instances of a service), consider the Histogram
// metric type. See the Prometheus documentation for more details.
//
// To create Summary instances, use NewSummary.
```

Reply:
Yes, you're right. I fixed it. Thanks!